src/middlewares/checkArticleOwnership.js

```diff
@@ -0,0 +1,48 @@
+import models from '../api/models';
+import statusCode from '../helpers/constants/status.codes';
+
+const { Article, User } = models;
+/**
+ * containing all user's model controllers (signup, login)
+ *
+ * @export
+ * @class CheckUserOwnership
+ */
+class CheckUserOwnership {
+  /**
+   * this is a middleware which checks if the user is registered in to our database.
+   *
+   * @author Frank Mutabazi
+   * @static
+   * @param {object} req the request
+   * @param {object} res the response to be sent
+   * @param { object } next the next route controller to be called
+   * @memberof CheckUserOwnership
+   * @returns {Object} res
+   */
+  static async checkOwner(req, res, next) {
+    const { user: { email } } = req.user;
+    const { slug } = req.params;
+    const result = await User.findOne({
```

Comment:
This work is done by the authorization middleware. Please, check it out and use it @MCFrank16